File: python/198_house-robber.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        if not nums:
            return 0

        dp = [0] * (len(nums) + 1)
        dp[1] = nums[0]

        for i in range(2, len(dp)):
            dp[i] = max(dp[i - 1], dp[i - 2] + nums[i - 1])

        return dp.pop()

File: go/198_house-robber.go
func rob(nums []int) int {
	dp := make([]int, len(nums)+1)
	dp[1] = nums[0]

	for i := 2; i < len(dp); i++ {
		dp[i] = maxInt(dp[i-1], dp[i-2]+nums[i-1])
	}
	return dp[len(dp)-1]
}

func maxInt(a, b int) int {
	if a > b {
		return a
	} else {
		return b
	}
}